embedded-workshop-book/src/radio-puzzle-help.md

```diff
@@ -0,0 +1,67 @@
+# Help
+
+## Use a dictionary.
+
+Our suggestion is to use a dictionary / map. `std::collections::HashMap` is not available in `no_std` code (without linking to a global allocator) but you can use one of the stack-allocated maps in the [`heapless`] crate. It supplies a stack-allocated, fixed-capacity version of the `std::Vec` type which will come in handy to store byte arrays. To store character mappings we recommend using a `heapless::IndexMap`.
+
+`heapless` is already declared as a dependency in the Cargo.toml of the project so you can directly import it into the application code using a `use` statement.
+
+[`heapless`]: https://docs.rs/heapless
+[crates.io]: https://crates.io/crates/heapless
+
+
+``` rust
+use heapless::Vec;         // like `std::Vec` but stack-allocated
+use heapless::FnvIndexMap; // a dictionary / map
+use heapless::consts::*;   // defines U16, U32, U64... etc. to set the size of the IndexMap
+
+fn main() {
+    // A hash map with a capacity of 16 key-value pairs allocated on the stack
+    let mut my_map = FnvIndexMap::<_, _, U16>::new();
+    my_map.insert(b'A', b'~').unwrap();
+
+    // A vector with a fixed capacity of 8 elements allocated on the stack
+    let mut my_vec = Vec::<_, U8>::new();
+    my_vec.push(b'A').unwrap();
+}
+```
+
+If you haven't used a stack-allocated collection before note that you'll need to specify the capacity of the collection as a type parameter using one of the "type-level values" in the `heapless::consts` module. The [`heapless::IndexMap` documentation][indexMap] of the `heapless` crate has some usage examples, as does the [`heapless::Vec` documentation][vec].
```

Comment:
I remember that it took me a while to understand that `U16` is a `heapless` constant, not a Rust quirk– would it make sense to clarify this by writing something like "...type parameter using one of the "type-level values" in the `heapless::consts` module (e.g. `U8`, `U64` etc)."?

Reply:
doesn't hurt to add a small list of examples, imo

Reply:
Lotte, do you have time to do that?

Reply:
sure, 1sec